core/src/replay_stage.rs

```diff
@@ -1936,7 +1936,7 @@ impl ReplayStage {
                 .map_or(false, |sender| sender.should_send_parents)
                 .then(|| {
                     let mut new_chain = rooted_slots.clone();
-                    new_chain.push(oldest_parent.unwrap_or(bank.parent_slot()));
+                    new_chain.push(oldest_parent.unwrap_or_else(|| bank.parent_slot()));
```

Comment:
I think this is due to the rust/clippy version of 1.14 vs master.

Reply:
Yes, well, I figured that was a given. I was hoping you would find out specifically what changed, since I recently ran into a v1.14/master clippy mismatch that was because we stopped CI checking for a legitimate lint: https://github.com/solana-labs/solana/pull/31690

But I've already done the legwork on this one: https://github.com/rust-lang/rust-clippy/blob/master/CHANGELOG.md#moves-and-deprecations-2
> Moved [or_fun_call](https://rust-lang.github.io/rust-clippy/master/index.html#or_fun_call) to nursery (Now allow-by-default) https://github.com/rust-lang/rust-clippy/pull/9829